src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs

```diff
@@ -502,7 +504,7 @@ public bool SetUserAgent(string productName, string version)
         /// <returns></returns>
         private string CleanUserAgentInfoEntry(string infoEntry)
         {
-            Regex pattern = new Regex("[~`!@#$%^&*(),<>?{} ]");            
+            Regex pattern = new Regex("[~`!@#$%^&*(),<>?{} \/]");            
```

Comment:
@ruant can you provide a scenario or an example where you are encountering an error?
Also add a test that covers the "/" scenario that you are encountering.

Reply:
```
System.FormatException: The format of value '6.1.12+Branch.release/6.1.12.Sha.3d0b9f3f9bb053111ba468326221f21b95124a3c' is invalid.
   at System.Net.Http.Headers.HeaderUtilities.CheckValidToken(String value, String parameterName)
   at System.Net.Http.Headers.ProductHeaderValue..ctor(String name, String version)
   at Microsoft.Rest.ServiceClient`1.InitializeHttpClient(HttpClientHandler httpClientHandler, DelegatingHandler[] handlers)
```

I'll whip up a test that covers it.